elastic/apm-server
# model/modeldecoder: parse transaction.sample_rate

## Motivation/summary

Look for the recently introduced transaction.sample_rate
property, and use it to set Transaction.RepresentativeCount.

If sample_rate is zero then we set Transaction.RepresentativeCount
to zero, and otherwise we set it to the inverse of sample_rate.

If sample_rate is not sent, Transaction.RepresentativeCount is set
to 1 as we assume the agent will send both sampled and unsampled
transactions, in which case all events should be counted equally.

See also: https://github.com/elastic/apm/blob/master/specs/agents/tracing-sampling.md#effect-on-metrics

## Checklist


I have considered changes for:
~- [ ] documentation~
~- [ ] logging (add log lines, choose appropriate log selector, etc.)~
~- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~
~- [ ] telemetry~
~- [ ] Elasticsearch Service (https://cloud.elastic.co)~
~- [ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)~
~- [ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)~

## How to test these changes

- Instrument an app with the Go agent, set sampling rate to something low like 1%
- Run server with transaction metrics enabled
- Wait for transaction metric docs to be indexed
- Check that transaction metric counts are approximately correct

## Related issues

Closes #4188
